src/services/players/players.service.ts

```diff
@@ -57,16 +62,40 @@ class PlayersService {
   };
 
   updatePlayerVehicle = (id: string, vehicleId: string, update: PlayerVehicleUpdate) => {
-    // check player exists
-    // check vehicle exists and belongs to player
-    // update vehicle
-    // return updated vehicle
+    const player = this.getPlayer(id);
+    let playerVehicle: PlayerVehicle = vehicles.find(
+      (vehicle) => vehicle.id === vehicleId && vehicle.playerId === player.id,
+    ) as PlayerVehicle;
+
+    if (playerVehicle) {
+      if (update.paintIndex !== undefined) {
+        playerVehicle.paintIndex = update.paintIndex;
+      }
+      if (update.resaleValue !== undefined) {
+        playerVehicle.resaleValue = update.resaleValue;
+      }
+    }
+
+    return playerVehicle;
   };
 
   updatePlayerSettings = (id: string, update: PlayerSettingsUpdate) => {
-    // check player exists
-    // update settings
-    // return updated settings
+    const player = this.getPlayer(id);
+    let playerSettings = this.getPlayerSettings(id);
+
+    if (playerSettings !== null) {
```

Comment:
it won't be null, so you can do either one of these suggestions but null won't work